docs/tempo/website/configuration/tls.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Configure TLS Communication
+weight: 70
+---
+
+# Configure TLS Communication
+
+Tempo can be configured to communicate between the components using Transport Layer Security, or TLS.
+
+Please note, the ciphers and TLS version here are for example purposes only. We are in no way recommending which ciphers or TLS versions for use in production environments, and are here for example purposes only.
+
+## Server Configuration
+
+```yaml
+server:
+  tls_cipher_suites: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
+  tls_min_version: VersionTLS12
+
+  grpc_tls_config:
+    cert_file: /tls/tls.crt
+    key_file: /tls/tls.key
+    client_auth_type: VerifyClientCertIfGiven
+    client_ca_file: /tls/ca.crt
+  http_tls_config:
+    cert_file: /tls/tls.crt
+    key_file: /tls/tls.key
+    client_auth_type: VerifyClientCertIfGiven
+    client_ca_file: /tls/ca.crt
+```
+
+Valid values for the `client_auth_type` are documented in the standard `crypt/tls` package under `ClientAuthType` [here](https://pkg.go.dev/crypto/tls#ClientAuthType).
+
+## Client Configuration
+
+Several components of Tempo need to configure the gRPC clients they use to communicate with other components. For example, when the `querier` contacts the `query-frontend` to request work, the client in use must enable TLS if the server is serving a TLS endpoint.
+
+The Tempo configuration uses a standard configuration stanza for each of these client configurations. Below is an example of the configuration.
+
+The optional configuration elements `tls_min_version`, `tls_cipher_suites`, and `tls_insecure_skip_verify` may be omitted. The option `tls_server_name` may or may not be required, depending on the environment.
+
+```yaml
+grpc_client_config:
+  tls_enabled: true
+  tls_cert_path: /tls/tls.crt
+  tls_key_path: /tls/tls.key
+  tls_ca_path: /tls/ca.crt
+  tls_server_name: tempo.trace.svc.cluster.local
+  tls_insecure_skip_verify: false
+  tls_cipher_suites: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
+  tls_min_version: VersionTLS12
+```
+
+The above configuration block needs to be set at the following configuration locations.
```

Comment:
I'm wondering if we should write this out as a procedure? When you say that the configuration block needs to be set, does that mean the entire code block should be included under each of the three sections listed below? And if so, are TLS options copied and pasted without any changes? (i.e., are there options that would be changed for any of the sections? Reading through, I don't think so, but I'm asking for clarity.)
```suggestion
The configuration block needs to be set at the following configuration locations.
```

Reply:
That's right.  The block would appear at each of the locations, unmodified.  Unless there was a reason to modify perhaps.  I can think of a couple reasons maybe that the server block has a more loose configuration on the query-frontends to support more clients or some such.